### PR TITLE
Add missing Firestore comments

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Add missing comments for Firestore's Public API and modify the API generation script to pull JSDoc comments from a hidden superclass to the new location.

### CODE REVIEW

File: packages/firestore/src/exp/reference_impl.ts [hunk 1]
```diff
@@ -417,7 +417,11 @@ export function addDoc<T>(
   return executeWrite(firestore, [mutation]).then(() => docRef);
 }
 
+/**
+ * A function returns by `onSnapshot() `that removes the listener when invoked.
```
Comment (egilmorez) [inline packages/firestore/src/exp/reference_impl.ts:421]: Suggest "returned by `onSnapshot()`

File: packages/firestore/src/exp/settings.ts [hunk 2]
```diff
@@ -19,10 +19,34 @@ import { Settings as LiteSettings } from '../lite/settings';
 
 export { DEFAULT_HOST } from '../lite/settings';
 
+/**
+ * Settings that can be passed to Firestore.enablePersistence() to configure
```
Comment (egilmorez) [inline packages/firestore/src/exp/settings.ts:23]: Backticks for literal.

File: packages/firestore/src/exp/settings.ts [hunk 3]
```diff
@@ -19,10 +19,34 @@ import { Settings as LiteSettings } from '../lite/settings';
...
 export interface Settings extends LiteSettings {
+  /**
+   * An approximate cache size threshold for the on-disk data. If the cache
+   * grows beyond this size, Firestore will start removing data that hasn't been
+   * recently used. The size is not a guarantee that the cache will stay below
+   * that size, only that if the cache exceeds the given size, cleanup will be
+   * attempted.
+   *
+   * The default value is 40 MB. The threshold must be set to at least 1 MB, and
+   * can be set to CACHE_SIZE_UNLIMITED to disable garbage collection.
```
Comment (egilmorez) [inline packages/firestore/src/exp/settings.ts:49]: Backticks for literal.

File: packages/firestore/src/lite/field_value.ts [hunk 4]
```diff
@@ -25,9 +25,11 @@ import { FieldTransform } from '../model/mutation';
 export abstract class FieldValue {
   /**
    * @param _methodName - The public API endpoint that returns this class.
+   * @hideconstructor
    */
   constructor(public _methodName: string) {}
 
+  /** Compares FieldValues for equality. */
```
Comment (egilmorez) [inline packages/firestore/src/lite/field_value.ts:32]: Backticks for literal.

File: packages/firestore/src/lite/reference.ts [hunk 5]
```diff
@@ -139,15 +141,17 @@ export class DocumentReference<T = DocumentData> {
...
-   * `null` removes the current converter.
+   * @param converter - Converts objects to and from Firestore.
    * @returns A `DocumentReference<U>` that uses the provided converter.
    */
-  withConverter(converter: null): DocumentReference<DocumentData>;
   withConverter<U>(converter: FirestoreDataConverter<U>): DocumentReference<U>;
+  /**
+   * Removes the current converter.
+   *
+   * @param converter - `null'`removes the current converter.
```
Comment (egilmorez) [inline packages/firestore/src/lite/reference.ts:151]: Looks like an extra "'" crept in there.

**Review by Feiyang1 — APPROVED** [review]

**Comment by Feiyang1** [comment]
Pending Eric's review

File: packages/firestore/src/lite/reference.ts [hunk 6]
```diff
@@ -180,6 +184,13 @@ export class Query<T = DocumentData> {
     this.firestore = firestore;
   }
 
+  /**
+   * Removes the current converter.
+   *
+   * @param converter - `null'`removes the current converter.
```
Comment (egilmorez) [inline packages/firestore/src/lite/reference.ts:190]: Here too, unless it really is null'

File: packages/firestore/src/lite/reference.ts [hunk 7]
```diff
@@ -251,10 +262,17 @@ export class CollectionReference<T = DocumentData> extends Query<T> {
...
    * @returns A `CollectionReference<U>` that uses the provided converter.
    */
-  withConverter(converter: null): CollectionReference<DocumentData>;
   withConverter<U>(
     converter: FirestoreDataConverter<U>
   ): CollectionReference<U>;
+  /**
+   * Removes the current converter.
+   *
+   * @param converter - `null'`removes the current converter.
```
Comment (egilmorez) [inline packages/firestore/src/lite/reference.ts:271]: Here too.

File: packages/firestore/src/lite/settings.ts [hunk 8]
```diff
@@ -28,12 +28,61 @@ import { validateIsNotUsedTogether } from '../util/input_validation';
...
+   */
   cacheSizeBytes?: number;
+
+  /**
+   * Forces the SDK’s underlying network transport (WebChannel) to use
+   * long-polling. Each response from the backend will be closed immediately
+   * after the backend sends data (by default responses are kept open in
+   * case the backend has more data to send). This avoids incompatibility
+   * issues with certain proxies, antivirus software, etc. that incorrectly
+   * buffer traffic indefinitely. Use of this option will cause some
```
Comment (egilmorez) [inline packages/firestore/src/lite/settings.ts:68]: Global nit: don't use "will" unless you really mean future tense.  In places like this, present tense is fine.  "This option causes some performance..."

**Review by egilmorez — COMMENTED** [review]
Some nits for you, thanks!

**Comment by schmidt-sebastian** [comment]
@egilmorez Can you take another look?

**Review by egilmorez — APPROVED** [review]
LGTM, thanks!

**Pull request merged**